tap_hubspot/__init__.py

```diff
@@ -229,13 +230,17 @@ def parse_source_from_url(url):
                       on_giveup=on_giveup,
                       factor=2)
 @utils.ratelimit(9, 1)
-def request(url, params=None):
+def request(url, params={}):
 
-    if CONFIG['token_expires'] is None or CONFIG['token_expires'] < datetime.datetime.utcnow():
-        acquire_access_token_from_refresh_token()
+    hapikey = CONFIG['hapikey']
+    if hapikey is None:
+        if CONFIG['token_expires'] is None or CONFIG['token_expires'] < datetime.datetime.utcnow():
+            acquire_access_token_from_refresh_token()
+        headers = {'Authorization': 'Bearer {}'.format(CONFIG['access_token'])}
+    else:
+        params = {**params, **{'hapikey': hapikey}}
```

Comment:
Why that vs `params['hapikey'] = hapikey`? This is very unidomatic and confusing.

Reply:
You are correct. I've made that amendment. 

My initial implementation had more than just the one `hapikey` key in the dictionary, hence the code to merge with `params`. This code was the product of a thought process that started with "how do I do this in Python 3?" and  spiraled to a confusing place.

Reply:
No worries. Merging dicts can be done with `.update`:

```
In [1]: params = {"a": 1, "b": 2}
   ...: params.update({"c": 3, "d": 4})
   ...: params
   ...:
Out[1]: {'a': 1, 'b': 2, 'c': 3, 'd': 4}
```